external-api/src/v1/controllers/number-generator.controller.ts

```diff
@@ -1,85 +1,92 @@
-// Number Generator API is used to get a new deal/facility id.
-// However, the id might already be in use so we need to check it with ACBS.
-//
-// the flow is:
-// 1) Call number generator Azure function
-// 2) return a PENDING status while Azure function runs independently
-// 3) Create a scheduled job to keep checking the Azure function status
-
-import axios from 'axios';
 import * as dotenv from 'dotenv';
+import axios, { HttpStatusCode } from 'axios';
 import { Request, Response } from 'express';
-import * as durableFunctionsLogController from './durable-functions-log.controller';
-
+import { NumberGeneratorResponse, NumberGeneratorErrorResponse } from '../../interfaces';
+import { InvalidEntityTypeError } from '../errors';
+import { ENTITY_TYPE, NUMBER_TYPE, USER } from '../../constants';
 dotenv.config();
-const numberGeneratorFunctionUrl = process.env.AZURE_NUMBER_GENERATOR_FUNCTION_URL;
 
-const callNumberGenerator = async ({ dealType, entityType, entityId, dealId, user }: any) => {
-  const response: any = await axios({
-    method: 'POST',
-    url: `${numberGeneratorFunctionUrl}/api/orchestrators/numbergenerator`,
-    data: {
-      dealType,
-      entityType,
-      entityId,
-      dealId,
-      user,
-    },
-  }).catch((error: any) => ({
-    error,
-  }));
+const { APIM_MDM_URL, APIM_MDM_VALUE, APIM_MDM_KEY } = process.env;
 
-  if (response.error) {
-    await durableFunctionsLogController.addDurableFunctionLog({
-      type: 'NUMBER_GENERATOR',
-      data: {
-        dealType,
-        entityType,
-        entityId,
-        dealId,
-        user,
-        error: await response.error.toJSON(),
-      },
-    });
+const headers = {
+  headers: {
+    [String(APIM_MDM_KEY)]: APIM_MDM_VALUE,
+    'Content-Type': 'application/json',
+  },
+};
 
-    return {
-      status: response.status || 500,
-    };
+/**
+ * Determines the number type based on the entity type.
+ * @param entityType - The type of entity for which the number type needs to be determined.
+ * @returns A number based on the value of entityType.
+ * @throws {Error} If entityType is invalid.
+ */
+export const getNumberTypeId = (entityType: string): number => {
+  switch (entityType) {
+    case ENTITY_TYPE.DEAL:
+      return NUMBER_TYPE.DEAL;
+    case ENTITY_TYPE.FACILITY:
+      return NUMBER_TYPE.FACILITY;
+    default:
+      throw new InvalidEntityTypeError(entityType);
   }
+};
+
+/**
+ * Retrieves a number from a number generator API based on the provided `entityType` and `dealId`.
+ * @param req - The HTTP request object containing the `entityType` and `dealId` in the `body` property.
+ * @param res - The HTTP response object used to send the response back to the client.
+ * @returns {Promise<Object>} The retrieved number in the response body.
+ */
+export const get = async (req: Request, res: Response): Promise<Response<NumberGeneratorResponse> | Response<NumberGeneratorErrorResponse>> => {
```

Comment:
Could we call this function something a bit more descriptive? Even if just 
`getIdForEntityFromNumberGenerator`